ibis/expr/operations/relations.py

```diff
@@ -640,26 +634,31 @@ class Aggregation(TableNode, sch.HasSchema):
                 )
             )
         ),
-        default=[],
+        default=(),
     )
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
-        if not self.by:
-            self.sort_keys.clear()
-
-    def _validate(self):
+    def __init__(self, table, metrics, by, having, predicates, sort_keys):
         from ibis.expr.analysis import FilterValidator
 
         # All non-scalar refs originate from the input table
-        all_exprs = self.metrics + self.by + self.having + self.sort_keys
-        self.table._assert_valid(all_exprs)
+        all_exprs = metrics + by + having + sort_keys
+        table._assert_valid(all_exprs)
 
         # Validate predicates
-        validator = FilterValidator([self.table])
-        validator.validate_all(self.predicates)
-
+        validator = FilterValidator([table])
+        validator.validate_all(predicates)
+
+        if not by:
+            sort_keys = tuple()
```

Comment:
I'm not following the logic here. `sort_keys` is an empty tuple if `by` is `False`-y?

Reply:
This was the previous implementation:

```python
if not self.by:
       self.sort_keys.clear()
```

I'm not exactly sure why we need it either.

Reply:
Ok, we can leave it for now